cli/lib/kontena/cli/stacks/yaml/reader.rb

```diff
@@ -147,10 +157,18 @@ def process_variables
         raise RuntimeError, "Variable validation failed: #{variables.errors.inspect}" unless variables.valid?
       end
 
+      # @raise [Liquid::Error]
       def interpolate_liquid(content, vars)
         Liquid::Template.error_mode = :strict
         template = Liquid::Template.parse(content)
-        template.render(vars, strict_variables: true, strict_filters: true)
+
+        # Wrap nil values in LiquidNull to not have Liquid consider them as undefined
+        vars = vars.clone
```

Comment:
There can be arrays as values. Maybe LiquidNull should be a subclass of string or something.

Hmm, I wonder how these look in the reader.execute[:variables] that gets sent to master. Maybe there needs to be some "deliquidnullify" after the interpolation?